provider/src/main/kotlin/org/gradle/kotlin/dsl/precompile/PrecompiledProjectScript.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.precompile
+
+import org.gradle.api.Project
+
+import org.gradle.kotlin.dsl.GradleDsl
+import org.gradle.kotlin.dsl.ScriptHandlerScope
+
+import org.gradle.plugin.use.PluginDependenciesSpec
+import org.gradle.plugin.use.PluginDependencySpec
+
+import kotlin.script.extensions.SamWithReceiverAnnotations
+import kotlin.script.templates.ScriptTemplateDefinition
+
+
+/**
+ * Script template definition for precompiled Kotlin script with a [Project] target.
+ */
+@ScriptTemplateDefinition(
+    resolver = PrecompiledScriptDependenciesResolver::class,
+    scriptFilePattern = "^.*\\.gradle\\.kts$")
```

Comment:
What about URL script with query parameters? #637
Or is this the wrong place to fix this?

Reply:
Wrong place to fix it and this template would never be applied to a remote script.

Reply:
Okay, I was confusing these with the applied scripts.